### PR TITLE
soc: nxp_imx: Move code/data location configs to soc level

### LINKED ISSUE #10271: Unexpected Kconfig warnings during documentation build
Building the documentation (``make htmldocs``)  is showing some new and unexpected warning messages:

```
warning: the default selection CODE_ITCM (defined at boards/arm/mimxrt1050_evk/Kconfig:13, boards/arm/mimxrt1060_evk/Kconfig:13) of <choice> (defined at boards/arm/mimxrt1050_evk/Kconfig:9) is not contained in the choice
warning: the choice symbol CODE_ITCM (defined at boards/arm/mimxrt1050_evk/Kconfig:13, boards/arm/mimxrt1060_evk/Kconfig:13) is defined with a prompt outside the choice
warning: the choice symbol CODE_HYPERFLASH (defined at boards/arm/mimxrt1050_evk/Kconfig:16, boards/arm/mimxrt1060_evk/Kconfig:16) is defined with a prompt outside the choice
warning: the choice symbol CODE_QSPI (defined at boards/arm/mimxrt1050_evk/Kconfig:19, boards/arm/mimxrt1060_evk/Kconfig:19) is defined with a prompt outside the choice
warning: the choice symbol CODE_ITCM (defined at boards/arm/mimxrt1050_evk/Kconfig:13, boards/arm/mimxrt1060_evk/Kconfig:13) is defined with a prompt outside the choice
warning: the choice symbol CODE_HYPERFLASH (defined at boards/arm/mimxrt1050_evk/Kconfig:16, boards/arm/mimxrt1060_evk/Kconfig:16) is defined with a prompt outside the choice
warning: the choice symbol CODE_QSPI (defined at boards/arm/mimxrt1050_evk/Kconfig:19, boards/arm/mimxrt1060_evk/Kconfig:19) is defined with a prompt outside the choice
```

### PR DESCRIPTION
**Pull request by MaureenHelm**
The imx rt family of socs has several options for linking code and data
into internal or external memories, and up until now we have handled
these options at the board level. This has resulted in several Kconfig
symbols being defined in multiple places and triggering warnings in
documentation builds:

warning: the default selection CODE_ITCM (defined at boards/arm/mimxrt1050_evk/Kconfig:13, boards/arm/mimxrt1060_evk/Kconfig:13) of <choice> (defined at boards/arm/mimxrt1050_evk/Kconfig:9) is not contained in the choice
warning: the choice symbol CODE_ITCM (defined at boards/arm/mimxrt1050_evk/Kconfig:13, boards/arm/mimxrt1060_evk/Kconfig:13) is defined with a prompt outside the choice
warning: the choice symbol CODE_HYPERFLASH (defined at boards/arm/mimxrt1050_evk/Kconfig:16, boards/arm/mimxrt1060_evk/Kconfig:16) is defined with a prompt outside the choice
warning: the choice symbol CODE_QSPI (defined at boards/arm/mimxrt1050_evk/Kconfig:19, boards/arm/mimxrt1060_evk/Kconfig:19) is defined with a prompt outside the choice
warning: the choice symbol CODE_ITCM (defined at boards/arm/mimxrt1050_evk/Kconfig:13, boards/arm/mimxrt1060_evk/Kconfig:13) is defined with a prompt outside the choice
warning: the choice symbol CODE_HYPERFLASH (defined at boards/arm/mimxrt1050_evk/Kconfig:16, boards/arm/mimxrt1060_evk/Kconfig:16) is defined with a prompt outside the choice
warning: the choice symbol CODE_QSPI (defined at boards/arm/mimxrt1050_evk/Kconfig:19, boards/arm/mimxrt1060_evk/Kconfig:19) is defined with a prompt outside the choice

The number of warnings increased as we added more imx rt boards. Fix the
warnings by moving code and data location configs from the board level
to the soc level.

The default memories for all imx rt boards are unchanged. The
mimxrt10{20,50,60}_evk boards still default to hyperflash/qspi for code
and sdram for data. The mimxrt1064_evk board still defaults to ITCM for
code and DTCM for data because jlink does not yet support programming
internal flash.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

Fixes #10271

### CODE REVIEW

**Review by dbkinder — APPROVED** [review]
bless you :)  Need to fix/remove the long lines in the commit message though...

File: soc/arm/nxp_imx/rt/Kconfig.soc [hunk 1]
```diff
@@ -246,4 +246,30 @@ config DEVICE_CONFIGURATION_DATA
...
+
+config CODE_FLEXSPI
+	bool "Link code into external FlexSPI-controlled memory"
+	select NXP_IMX_RT_BOOT_HEADER
+
+endchoice
+
+choice DATA_LOCATION
+	prompt "Data location selection"
+	default DATA_SEMC
```
Comment (aurel32) [inline soc/arm/nxp_imx/rt/Kconfig.soc]: Is it fine defaulting to SDRAM at the SoC level? All the EVK boards have SDRAM, but not all boards necessary have one.

**Review by mbolivar — APPROVED** [review]
Thank you! I've tested and verified the warnings go away.

File: soc/arm/nxp_imx/rt/Kconfig.soc (hunk 1 above)
Comment (MaureenHelm) [reply soc/arm/nxp_imx/rt/Kconfig.soc]: My original thought was the most common case should be the default, but you make a good point about non-EVK boards. Will rework the defaults such that TCM is default at the SoC level and EVK boards override them.

**Comment by MaureenHelm** [comment]
> Need to fix/remove the long lines in the commit message though...

I knew gitlint would flag that, but I felt it was informational enough to override the long line rule in this case.

**Review by nashif — APPROVED** [review]

**Pull request merged**